src/db/sysdb.c

```diff
@@ -2106,7 +2106,6 @@ bool sysdb_entry_attrs_diff(struct sysdb_ctx *sysdb,
     if (res->count == 0) {
         return true;
     } else if (res->count != 1) {
-        ret = EIO;
```

Comment:
The only place where "ret" variable is used is line 2102. Maybe instead of removing "ret" assignment here it is good idea to get ride of "ret" variable at all? And for this single DEBUG() call use sysdb_error_to_errno() directly?

Reply:
I think I once proposed such an "optimization" but it was refused with a wording "having explicit variable helps to set gdb breakpoint".
So seems like kind of coding style.

Reply:
This may be true, but still - in case of this function "lret" variable can be a hook for gdb attach. In fact "ret" is just used to convert "lret" value for debug log purposes only. The main information about function execution is still stored in "lret" value.

Reply:
Well, this is largely coding style question.

From my point of view, this PR can be accepted "as is" because it does what it claims to, i.e. resolves coverity issue.

Further improvement, if required, could be done in a separate PR.

Reply:
I don't mind changing the PR to include Pawel's improvement, but I'd like the coding style clarified first. Who can clarify it?

Reply:
The unwritten rule is that more complex functions should not be nested in other calls to keep stepping in gdb clear. But this case is equivalent to `strerror` or `sss_strerror` so I'm personally fine either way.

Reply:
Taking into account all the comments I'll update the code to remove ```ret``` and nest the call.